### DISCUSSION TITLE
Support for Weight Training?

### DISCUSSION THREAD

**Discussion by MeGodzillaUDead** [ssaring/sportstracker#258] [Ideas]
Hello, SportsTracker is great!  I am interested in also using it for tracking weight lifting.  It would be good to have at least the number of sets as well as the number of reps and amount of weight/resistance for each set. 

I do not have much experience with Java, but I would like learn more by helping with this project!

**Comment by ssaring** [comment]
Have already thought about that a while ago, but IMHO it does not provide much value when just adding the number of sets, reps and amount of weight. There should also be a selection for the type of exercise etc. 
Also from statistics side, this does not fit well with the existing statistics for the endurance related sport types, another kind of statistics would be needed as well. Therefore I decided back then not to invest the needed effort and not to make the application more complex then needed just for that.
If weight lifting is your primary sport type, then SportsTracker might not be the best fitting app for that. If it's just a side sport beside other endurance sport types, then you might create a Weight Lifting sport type and add the exercise details in the comment window as a lightweight workaround. At least you could track then the execution of that exercises, the duration, HR, calories etc.

**Comment by MeGodzillaUDead** [comment]
Thanks for the response.  I understand your decision, but I do not see
things the same way.  I really like the interface you have and think it is
worth adding the new sport type.  If you feel like giving any hints to
where in the code I can look that would be great, but I certainly don't
expect you to put any effort into something you don't see value in.

**Comment by ssaring** [comment]
Please don't misunderstand me, it's not that I don't see any value in there. Related to the needed effort the value seems to low at least for my usages. I also always favour to keep things simplistic and therefore question all special new requirements whether they meet the requirements of the majority of users. That's my thoughts on that.

If you would like to dig into in your own close, you should take first a look at the model class Exercise.java, the related ExerciseDialog classes and the persistence layer for storing all the new properties in database. Then you can focus on evaluation capabilities.
